app/views/LocationController.js

```diff
@@ -13,102 +13,90 @@ import { SvgXml } from 'react-native-svg';
 import settingsIcon from './../assets/svgs/settingsIcon';
 import { isPlatformAndroid } from './../Util';
 import Colors from '../constants/colors';
-import LocationServices from '../services/LocationService';
-import DefaultPage from './tracking/DefaultPage';
-import ExposurePage from './tracking/ExposurePage';
-import OffPage from './tracking/OffPage';
-import styles from './tracking/style';
-import UnknownPage from './tracking/UnknownPage';
+import LocationServices, { Reason } from '../services/LocationService';
+import { DefaultPage } from './tracking/DefaultPage';
+import { ExposurePage } from './tracking/ExposurePage';
+import { OffPage } from './tracking/OffPage';
+import { styles } from './tracking/style';
+import { UnknownPage } from './tracking/UnknownPage';
 
-export default class LocationController extends Component {
-  constructor(props) {
-    super(props);
-
-    if (isPlatformAndroid()) {
-      StatusBar.setBackgroundColor(Colors.TRANSPARENT);
-      StatusBar.setBarStyle('light-content');
-      StatusBar.setTranslucent(true);
-    }
-
-    this.state = {
-      location: {
-        canTrack: true,
-        reason: '',
-        crossedPaths: false,
-      },
-    };
-  }
-
-  componentDidMount() {
-    this.updateStateInfo();
-    // refresh state if user backgrounds app
-    AppState.addEventListener('change', () => {
-      this.updateStateInfo();
-    });
-    // refresh state if settings change
-    this.unsubscribe = this.props.navigation.addListener('focus', () => {
-      this.updateStateInfo();
-    });
-    // handle back press
-    BackHandler.addEventListener('hardwareBackPress', this.handleBackPress);
+export const LocationController = props => {
+  if (isPlatformAndroid()) {
+    StatusBar.setBackgroundColor(Colors.TRANSPARENT);
```

Comment:
Is this imperative stuff needed? I think you can do it in the render block as JSX: `<StatusBar backgroundColor=x />`

If actually needed here as a side effect, add a comment why, and put all these bits in the main `useEffect` block.